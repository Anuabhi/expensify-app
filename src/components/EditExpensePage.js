import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {removeExpense} from '../actions/expenses';
import {editExpense,addExpense} from '../actions/expenses';
//import expenses from '../tests/fixture/expenses';

export class EditExpensePage extends React.Component{
    onSubmit=(expense)=>{
        this.props.editExpense(this.props.id,expense);
        this.props.history.push('/');
    }


    onClick=()=>{
        this.props.removeExpense({id:this.props.id})
        this.props.history.push('/');
    }




render(){
    
    return(
    <div>
    <ExpenseForm
     expense={this.props.expense}
     onSubmit={this.onSubmit} />
     
     <button onClick={this.onClick}>Remove</button>

     Editing the Expense page
    </div>
);
}
}


const mapDispatchToProps=(dispatch)=>({
    
    editExpense:(id,updates)=>dispatch(editExpense(id,updates)),
    removeExpense:({id})=>dispatch(removeExpense({id}))


})

const mapStateToProps=(state,props)=>{
    return{
        expense:state.expenses.find((expense)=>
            expense.id===props.match.params.id
        )
    }
}
export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);