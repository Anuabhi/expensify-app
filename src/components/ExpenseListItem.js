import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';
import { NavLink } from 'react-router-dom';

const ExpenseListItem =({id,description,amount,createdAt}) =>(
    <div>
    <NavLink to={`/edit/${id}`}>{description}</NavLink>
    <p>{amount}</p>
    <p>{createdAt}</p>
    
    </div>
    
)

export default ExpenseListItem;
