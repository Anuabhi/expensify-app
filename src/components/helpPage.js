import React from 'react';
const helpPage =() =>(
    <div>
                This is from help component
    </div>
);
export default helpPage;