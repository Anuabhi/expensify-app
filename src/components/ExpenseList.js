import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/getVisibleExpenses';


export const ExpenseList =(props) =>(
    <div>
    

    
    
     <div>
     {
         props.expenses.length===0?(<p>Noexpenses</p>):(
     
      props.expenses.map((expense)=>(
    
       <ExpenseListItem  key={expense.id} {...expense}/>)
    ))
   }
    </div>
    </div>
)

const mapStateToProps=(state)=>{
    return{
        expenses:selectExpenses(state.expenses,state.filters)
    }
}



export default connect(mapStateToProps)(ExpenseList);

