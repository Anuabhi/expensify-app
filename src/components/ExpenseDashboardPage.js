import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseList from './ExpenseList';
import ExpenseListItem from './ExpenseListItem';
import ExpenseListFilters from './ExpenseListFilters';



const ExpenseDashboardPage =() =>(
    <div>
    This is from my Dashboard components
    

    <ExpenseListFilters />


    <ExpenseList />
    
    </div>

);

export default ExpenseDashboardPage;