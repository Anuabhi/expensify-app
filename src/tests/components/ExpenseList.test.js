import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixture/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render ExpenseList with expenses',()=>{
    const wrapper=shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
})


test('should render',()=>{
    const wrapper=shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
})



