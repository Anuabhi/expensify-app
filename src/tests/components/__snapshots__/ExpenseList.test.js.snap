// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div>
    <p>
      Noexpenses
    </p>
  </div>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <div>
    <ExpenseListItem
      amount={300}
      createdAt={300}
      description="this is biblo"
      id="1"
      key="1"
      note="I am gonna kill u :p  from sauroman"
    />
    <ExpenseListItem
      amount={400}
      createdAt={-345600000}
      description="this is frodo"
      id="2"
      key="2"
      note="hey ::: from Gandalf"
    />
    <ExpenseListItem
      amount={500}
      createdAt={-518400000}
      description="this is Peter"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;
